src/refiners/foundationals/fuyu/fuyu.py

```diff
@@ -0,0 +1,267 @@
+import re
+from dataclasses import dataclass, replace
+from typing import Any, List, Union
+
+import numpy as np
+import torch
+from PIL import Image
+from torch import Tensor, device as Device, dtype as DType
+
+import refiners.fluxion.layers as fl
+from refiners.fluxion.utils import no_grad
+from refiners.foundationals.fuyu.input_processor import InputEncoder
+from refiners.foundationals.fuyu.tokenizer import FuyuTokenizer
+from refiners.foundationals.fuyu.transformers import FuyuTransformer, FuyuTransformerLayer
+
+
+@dataclass(frozen=True)
+class Fuyu8b:
+    """
+    config with the base argument for Fuyu8b
+    """
+
+    embedding_dim: int = 4_096
+    feedforward_dim: int = 16_384
+    max_sequence_length: int = 16_384
+    vocabulary_size: int = 262_144
+    tokenizer: FuyuTokenizer = FuyuTokenizer()
+    patch_size: int = 30
+    padding_value: float = 1.0 / 255
+    num_layers: int = 36
+    num_heads: int = 64
+    norm_eps: float = 1e-5
+    base: int = 25_000
+    partial_rotary_factor: float = 0.5
+    use_bias: bool = True
+    is_optimized: bool = False
+    device: Device | str | None = "cuda:0"
+    dtype: DType | None = torch.float32
+
+    def with_device(self, new_device: Union[Device, str]) -> "Fuyu8b":
+        """
+        Returns a new instance of Fuyu8b with a specified device.
+
+        Args:
+            new_device (Union['Device', str]): The device to set for the new instance.
+
+        Returns:
+            Fuyu8b: New instance with updated device.
+        """
+        return replace(self, device=new_device)
+
+
+def create_fuyu(config: Fuyu8b):
+    """
+    create a fuyu model based on the config provided
+
+    Example:
+        ```py
+        config = Fuyu8b
+        network = create_fuyu(config)
+        ```
+    """
+    model = Fuyu(
+        embedding_dim=config.embedding_dim,
+        feedforward_dim=config.feedforward_dim,
+        max_sequence_length=config.max_sequence_length,
+        vocabulary_size=config.vocabulary_size,
+        tokenizer=config.tokenizer,
+        patch_size=config.patch_size,
+        padding_value=config.padding_value,
+        num_layers=config.num_layers,
+        num_heads=config.num_heads,
+        norm_eps=config.norm_eps,
+        base=config.base,
+        partial_rotary_factor=config.partial_rotary_factor,
+        use_bias=config.use_bias,
+        is_optimized=config.is_optimized,
+        device=config.device,
+        dtype=config.dtype,
+    )
+    return model
```

Comment:
no need to make config dataclasses, we'll simply use the constructor of the Fuyu class.
though you could declare a Fuyu8b class which is "pre-filled" with the above config, a bit like how it's done with DINOv2.

Reply:
Created a pre-filled Fuyu8b class in which tokenizer_path, device and dtype need to be specified.